### PR TITLE
Check if events must be on main

### LINKED ISSUE #2242: Doublecheck if Events must be sent on Main
See https://github.com/deltachat/deltachat-ios/pull/2236#issuecomment-2239155737

### PR DESCRIPTION
**Pull request by zeitschlag**
More minor refactorings. I checked every notification posted in `events.swift` if posting on main was necessary and it turned out: Most of the stuff gets already handled on main (except one or two, I fixed that).

Closes #2242.

### CODE REVIEW

File: DcCore/DcCore/DC/events.swift [hunk 1]
```diff
@@ -56,118 +56,105 @@ public class DcEventHandler {
...
-            DispatchQueue.main.async {
-                NotificationCenter.default.post(name: Event.importExportProgress, object: nil, userInfo: [
-                    "progress": Int(data1),
-                    "error": Int(data1) == 0,
-                    "done": Int(data1) == 1000,
-                    "errorMessage": self.dcAccounts.get(id: accountId).lastErrorString,
-                ])
-            }
+            
+            NotificationCenter.default.post(name: Event.importExportProgress, object: nil, userInfo: [
```
Comment (r10s) [inline DcCore/DcCore/DC/events.swift:75]:
```suggestion
            NotificationCenter.default.post(name: Event.importExportProgress, object: nil, userInfo: [
```

(just normalising whitespace)

File: DcCore/DcCore/DC/events.swift [hunk 2]
```diff
@@ -56,118 +56,105 @@ public class DcEventHandler {
...
+                "progress": Int(data1),
+                "error": Int(data1) == 0,
+                "done": Int(data1) == 1000,
+                "errorMessage": self.dcAccounts.get(id: accountId).lastErrorString,
+            ])
         case DC_EVENT_MSGS_CHANGED:
```
Comment (r10s) [inline DcCore/DcCore/DC/events.swift:81]:
```suggestion
            ])

        case DC_EVENT_MSGS_CHANGED:
```

(just normalising whitespace)

File: DcCore/DcCore/DC/events.swift [hunk 3]
```diff
@@ -56,118 +56,105 @@ public class DcEventHandler {
...
-            DispatchQueue.main.async {
-                NotificationCenter.default.post(name: Event.chatModified, object: nil, userInfo: [
-                    "chat_id": Int(data1),
-                ])
-            }
+
+            NotificationCenter.default.post(name: Event.chatModified, object: nil, userInfo: [
+                "chat_id": Int(data1),
+            ])
         case DC_EVENT_CHAT_EPHEMERAL_TIMER_MODIFIED:
```
Comment (r10s) [inline DcCore/DcCore/DC/events.swift:119]:
```suggestion
            ])

        case DC_EVENT_CHAT_EPHEMERAL_TIMER_MODIFIED:
```

(just normalising whitespace)

File: DcCore/DcCore/DC/events.swift [hunk 4]
```diff
@@ -56,118 +56,105 @@ public class DcEventHandler {
...
+                "contact_id": Int(data1)
+            ])
 
         case DC_EVENT_CONNECTIVITY_CHANGED:
             if accountId != dcAccounts.getSelected().id {
                 return
             }
             logger.info("📡[\(accountId)] connectivity changed")
-            DispatchQueue.main.async {
                 NotificationCenter.default.post(name: Event.connectivityChanged, object: nil)
```
Comment (r10s) [inline DcCore/DcCore/DC/events.swift:157]:
```suggestion
            NotificationCenter.default.post(name: Event.connectivityChanged, object: nil)
```

(just normalising whitespace)

File: deltachat-ios/Controller/AccountSetup/Instand Onboarding/InstantOnboardingViewController.swift [hunk 5]
```diff
@@ -238,9 +238,11 @@ class InstantOnboardingViewController: UIViewController {
...
 
     private func handleCreateSuccess() {
-        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
-        appDelegate.registerForNotifications()
-        appDelegate.reloadDcContext()
+        DispatchQueue.main.async {
+            guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+            appDelegate.registerForNotifications()
+            appDelegate.reloadDcContext()
+        }
```
Comment (r10s) [inline deltachat-ios/Controller/AccountSetup/Instand Onboarding/InstantOnboardingViewController.swift:245]: took me a moment to verify that handleCreateSuccess() is called only from main, good catch!

File: deltachat-ios/Controller/Settings/SettingsViewController.swift [hunk 6]
```diff
@@ -242,7 +242,7 @@ internal final class SettingsViewController: UITableViewController {
         } else {
             NotificationManager.removeAllNotifications()
         }
-        UserDefaults.standard.synchronize()
```
Comment (r10s) [inline deltachat-ios/Controller/Settings/SettingsViewController.swift:245]: we can probably remove that safely as it was not called always anyways?

documentation says it is "deprecated" only, so not sure what it does in detail or why it was used in the past :)

**Review by r10s — APPROVED** [review]
nice! code lgtm, some minors.

i also did test some standard things and double-checked some "more important" code flows -  i did not run into issues yet

great job!

File: deltachat-ios/Controller/Settings/SettingsViewController.swift (hunk 6 above)
Comment (zeitschlag) [reply deltachat-ios/Controller/Settings/SettingsViewController.swift:245]: It doesn't do anything, documentation reads

> this method is unnecessary and shouldn't be used.

**Pull request merged**